web/packages/teleterm/src/ui/services/deepLinks/deepLinksService.ts

```diff
@@ -88,13 +88,70 @@ export class DeepLinksService {
         break;
       }
       case '/authenticate_web_device': {
-        // TODO (avatus): add authenticate device web confirmation dialog
-        // this case is a stub and will not be reached
+        const id = result.url?.searchParams?.id;
+        const token = result.url?.searchParams?.token;
+        await this.askAuthorizeDeviceTrust(result.url, id, token);
         break;
       }
     }
   }
 
+  private async askAuthorizeDeviceTrust(
+    url: DeepURL,
+    id: string,
+    token: string
+  ): Promise<void> {
+    const rootClusterId = url.hostname;
+    const clusterAddress = url.host;
+    const prefill = {
+      clusterAddress,
+      username: url.username,
+    };
+
+    const rootClusterUri = routing.getClusterUri({ rootClusterId });
+    const rootCluster = this.clustersService.findCluster(rootClusterUri);
+
+    if (!rootCluster) {
+      const { canceled } = await new Promise<{ canceled: boolean }>(resolve => {
+        this.modalsService.openRegularDialog({
+          kind: 'cluster-connect',
+          clusterUri: undefined,
+          reason: undefined,
+          prefill,
+          onSuccess: () => resolve({ canceled: false }),
+          onCancel: () => resolve({ canceled: true }),
+        });
+      });
+
+      if (canceled) {
+        return;
+      }
+    }
+
+    const { isAtDesiredWorkspace } =
+      await this.workspacesService.setActiveWorkspace(rootClusterUri, prefill);
+
+    if (!isAtDesiredWorkspace) {
+      return;
+    }
+
+    this.modalsService.openImportantDialog({
```

Comment:
I think we should open just a regular dialog here, especially since we're already using `openRegularDialog` above for the login modal.

I should probably update the comment for `openImportantDialog` to underline that it should be reserved for situations where a user might not expect that the action that triggered the modal would shift focus to Connect.

So for example when connecting to a db with a 3rd party program and expired certs, the user might not expect that this will bring up Connect. They might have some unfinished work in a regular modal in Connect, that's why we use the important one.

But in this case the user knows that they're about to open Connect.

It's not a huge deal overall, but I think we should be consistent with how we use it.